### ISSUE TITLE
Improve the registration flow

### ISSUE THREAD

**Issue by crazyyuan** [metagov/daostar#159]
Now the way to register DAO URI:
1. Treat registration as a proposal execution, website registration parameters need to be filled in in advance.
2. Registering on the website and in the contract is one process. We can improve it.
How: Add a button on the website registration result page to call the contract to complete the whole registration, which is so quickly.

**Comment by amanwithwings** [comment]
"Treat registration as a proposal execution" - @crazyyuan, do you mean generating a daoURI will now trigger an onchain transaction to publish it?

**Comment by crazyyuan** [comment]
> "Treat registration as a proposal execution" - @crazyyuan, do you mean generating a daoURI will now trigger an onchain transaction to publish it?

yes, it belongs to internal transactions, like this: https://etherscan.io/address/0x37df3fc47c1c3a2acafd2dad9c1c00090a8655bc#internaltx

this is 2th way:
https://optimistic.etherscan.io/address/0x5C0340AD34f7284f9272E784FF76638E8dDb5dE4

**Comment by amanwithwings** [comment]
My question would be: (1) often times, daoURIs are generated by a DAO contributor or a DAOstar contributor. If this person also publishes the daoURI onchain, how can we ensure that it is valid? The whole point is to have the DAO's main account (treasury/governance module) point to a daoURI.

**Comment by thelastjosh** [comment]
Ref. small ENS grant to explore this research?
